packages/langchain/src/orchestration/util.ts

```diff
@@ -174,3 +194,85 @@ export function mapOutputToChatResult(
     }
   };
 }
+
+/**
+ * Converts orchestration stream chunk to a LangChain message chunk.
+ * @param chunk- The orchestration stream chunk.
+ * @returns An {@link AIMessageChunk}
+ * @internal
+ */
+export function mapOrchestrationChunkToLangChainMessageChunk(
+  chunk: OrchestrationStreamChunkResponse
+): AIMessageChunk {
+  const { module_results, request_id } = chunk.data;
+  const content = chunk.getDeltaContent() ?? '';
+  const toolCallChunks = chunk.getDeltaToolCallChunks();
+
+  const additional_kwargs: Record<string, unknown> = {
+    module_results,
+    request_id
+  };
+
+  let tool_call_chunks: ToolCallChunk[] = [];
+  if (toolCallChunks) {
+    tool_call_chunks = mapOrchestrationToLangChainToolCallChunk(toolCallChunks);
+  }
+  // Use `AIMessageChunk` to represent message chunks for roles such as 'tool' and 'user' as well.
+  // While the `ChatDelta` type can accommodate other roles in the orchestration service's stream chunk response, in realtime, we only expect messages with the 'assistant' role to be returned.
+  return new AIMessageChunk({ content, additional_kwargs, tool_call_chunks });
+}
+
+/**
+ * Sets finish reason on a LangChain message chunk if available.
+ * @param messageChunk - The LangChain message chunk to update.
+ * @param finishReason - The finish reason from the response.
+ * @internal
+ */
+export function setFinishReason(
+  messageChunk: AIMessageChunk,
+  finishReason: string | undefined
+): void {
+  if (finishReason) {
+    messageChunk.response_metadata.finish_reason = finishReason;
+  }
+}
+
+/**
+ * Sets usage metadata on a message chunk if available.
+ * @param messageChunk - The LangChain message chunk to update.
+ * @param tokenUsage - The token usage information.
+ * @internal
+ */
+export function setTokenUsage(
+  messageChunk: AIMessageChunk,
+  tokenUsage: TokenUsage | undefined
+): void {
+  if (tokenUsage) {
+    messageChunk.usage_metadata = {
+      input_tokens: tokenUsage.prompt_tokens,
+      output_tokens: tokenUsage.completion_tokens,
+      total_tokens: tokenUsage.total_tokens
+    };
+    messageChunk.response_metadata.token_usage = tokenUsage;
+  }
+}
+
+/**
+ * Computes token indices for a chunk of the orchestration stream response.
+ * @param chunk - A chunk of the orchestration stream response.
+ * @returns An object with prompt and completion indices.
+ * @internal
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function computeTokenIndices(chunk: OrchestrationStreamChunkResponse): {
```

Comment:
[q] Didnt get the point of this function if its not supported in entirety. How is this helpful right now?

Reply:
Token indices need to be set in each generated chunk as this provides information about which prompt(in a multi-prompt scenario) and which choice(in a muti-choice scenario) the chunk belongs to.

Although, we hardcode these values now, leaving the function as-is enables us to support mutli-chat scenarios in the future easily.

Reply:
ok, is this in a ticket somewhere? I would add a TODO in the comments, but also mention it in a ticket for future.

Reply:
Added a BLI: https://github.com/SAP/ai-sdk-js-backlog/issues/321 and TODO.